libraries/libfc/include/fc/container/ordered_diff.hpp

```diff
@@ -39,48 +44,56 @@ class ordered_diff {
          if (s < source.size() && t < target.size()) {
             if (source[s] == target[t]) {
                // nothing to do, skip over
-               assert(s <= std::numeric_limits<SizeType>::max());
-               assert(t <= std::numeric_limits<SizeType>::max());
+               FC_ASSERT(s <= std::numeric_limits<SizeType>::max());
+               FC_ASSERT(t <= std::numeric_limits<SizeType>::max());
```

Comment:
Can we move these two `FC_ASSERT`  right after the `while` line instead of repeating them in almost every case?